NXP Kinetis K64F: Fixed bug in common fault handler and added CAN interrupts

jgrivera67 opened the pull request:
The subprogram to print the fault message on the UART
was sending characters to the UART without waiting for
previous transmits to finish.

Also, the common fault handler now prints to the UART the address
(in decimal) of the machine instruction that triggered the
fault.

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
NXP Kinetis K64F: Fixed bug in common fault handler

The subprogram to print the fault message on the UART
was sending characters to the UART without waiting for
previous transmits to finish.

Also, the common fault handler now prints to the UART the address
(in decimal) of the machine instruction that triggered the
fault.

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
NXP Kinetis K64F: Added CAN interrupts

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
Kinetis KL25Z: Added entries to interrupt vector table

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
Kinetis: Added section to linker scripts

Added section for code to be placed in SRAM.

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
NXP Kinetis KL25Z: Added version of s-musplo.adb to support Cortex-M0+

Also, some minor corrections.

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
NXP Kinetis K64F: Fixed path to linker scripts

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
Added memory protection unit support skeleton

Signed-off-by: German Rivera <jgrivera67@gmail.com>

lambourg commented:
Hi German,

This is just to let you know that I'm not ignoring the hard work you're doing on this pull request...

We (e.g AdaCore) are currently in the process of putting in place a CLA (Contributor License Agreement), that we'll need you to agree upon before accepting your contributions.

I know it's a bit administrative work, but the nice thing is once you agree with it once, we don't have to sign anything afterwards.

I'll keep you in touch as soon as I have more concrete information to give you, but still your current contribution is very appreciated, so please, accept my thanks :)

jgrivera67 commented:
Hi Jerome,

Thanks for your message. No problem, just let me know when the CLA is
available.

German


On Tue, Jan 24, 2017 at 2:33 PM, Lambourg Jérôme <notifications@github.com>
wrote:

> Hi German,
>
> This is just to let you know that I'm not ignoring the hard work you're
> doing on this pull request...
>
> We (e.g AdaCore) are currently in the process of putting in place a CLA
> (Contributor License Agreement), that we'll need you to agree upon before
> accepting your contributions.
>
> I know it's a bit administrative work, but the nice thing is once you
> agree with it once, we don't have to sign anything afterwards.
>
> I'll keep you in touch as soon as I have more concrete information to give
> you, but still your current contribution is very appreciated, so please,
> accept my thanks :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AdaCore/embedded-runtimes/pull/6#issuecomment-274929742>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEhcsB0Q5VppaKIjR4wZZ6lo29L8rVvRks5rVl_-gaJpZM4K7E6c>
> .
>

jgrivera67 committed:
Preliminary MPU support

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
Minor changes to MPU support

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
More MPU support

Signed-off-by: German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
MPU interface refactoring and added MPU support for DMA

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
MPU support complete

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
MPU support: Tasks run in privileged mode

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/AdaCore/embedded-runtimes?pullRequest=6) <br/>All committers have signed the CLA.

jgrivera67 committed:
Redesigned MPU support

Generalized task-private regions to be
Context-private regions instead. So that
ISRs can use the same primitives to set component,
parameter and MMIO regions.

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
MPU support working

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

jgrivera67 committed:
Fixed bug in Systick handler and soem refactoring

Signed-off-by: J. German Rivera <jgrivera67@gmail.com>

lambourg commented:
Hello German,

Thank you again for your contribution to this port.

As you may have noticed, the Contributors License Agreement is now ready for you to sign. You can sign it online to speed up the process, or send it back to us via regular mail.

As soon as this is done, I'll be able to merge your contribution.

Best regards,
- Jerome
